controllers/webhooks/component_webhook.go

```diff
@@ -66,21 +66,36 @@ func (r *ComponentWebhook) Default(ctx context.Context, obj runtime.Object) erro
 	if len(component.OwnerReferences) == 0 && component.DeletionTimestamp.IsZero() {
 		// Get the Application CR
 		hasApplication := appstudiov1alpha1.Application{}
-		err := r.client.Get(ctx, types.NamespacedName{Name: component.Spec.Application, Namespace: component.Namespace}, &hasApplication)
+		err := r.client.Get(context.Background(), types.NamespacedName{Name: component.Spec.Application, Namespace: component.Namespace}, &hasApplication)
```

Comment:
The update of the context here is related with the change of the `kubeconfig`? If yes should we make it more explicit in case we need to revisit this?